docfx_project/articles/bootstrapping-trust/part-03.certificates.md

```diff
@@ -0,0 +1,180 @@
+Public certificates are generated, signed, and distributed widely.
+Handling private certificates/keys is a sensitive matter and should be
+kept secure.
+
+Certificates contain more than keys. The Distinguished Name (DN) are
+text fields. They are useful mainly to humans to know what/who owns a
+certificate. It is sometimes used by software as display information or
+for comparison checks. Since humans provide the DN values or configure
+software with values, it is not always distinguishing. DN values have an
+alternate name: relatively distinguished names.
+
+Related to the Subject DN is the Issuer Information. The Issuer
+Information is the subject of the certificate that issued the
+certificate. Because of this, both the issuer information has similar
+values to the subject DN. Both can include the following information:
+
+- CN - common name - a name
+- SN - surname
+- SERIALNUMBER - a number that is usually unique per certificate issuer,
+  but not always
+- C - country
+- L - locality name
+- ST or S - state or province
+- STREET - street address
+- O - organization name
+- OU - organizational unit
+- T - title
+- G or GN - given name
+- E - email address
+- UID - user id
+- DC - domain component
+
+All, some, or none of these fields may be important. Whether they are or
+are not depends on the software where that will utilize them.
+
+#### Validity Period
+
+The Validity Period specifies two points in time from when the
+certificate is valid. Before and after this window of time, the
+certificate is invalid and should not be trusted. Validity periods
+should be as small as possible to fit the use case. Shorter periods
+reduce the window of time that compromised private key can remain useful
+for an attack. The cost of this is overhead reissuing certificates as
+they reach the end of their validity period.
+
+#### Usage/Extensions
+
+Usage/Extension Data is interesting because it can limit what roles a
+certificate fulfills. Depending on the system, this may be adhered to or
+not. Some examples of usage that are common to see:
+
+- key usage: client authentication, server authentication, signatures,
+  etc.
+- Subject Alternate Names (SANs)
+  - Limits what IP address, email address, domain name, etc. the
+    certificate can be associated with
+- Certificate Authority (CA) flag
+- and more
+
+This series will not dive into the details of these usages. However, it
+is essential to be aware of them and that they can affect the roles a
+certificate can play and that it is necessary to configure them
+correctly.
+
+#### Signatures
+
+The signature section of a certificate is a list of signatures from
+other entities that trust this certificate. A certificate that signs
+itself is a "self-signed certificate." Self-signed certificates must be
+individually trusted as no other certificate has expressed trust in it
+by signing it. Self-signed certificates sometimes used for testing
```

Comment:
typo - missing word?
"Self-signed certificates sometimes used for testing"
--> "Self-signed certificates are sometimes used for testing"

Reply:
fixed